src/ray/gcs/gcs_server/gcs_health_check_manager.h

```diff
@@ -108,8 +108,8 @@ class GcsHealthCheckManager : public std::enable_shared_from_this<GcsHealthCheck
   /// It can be updated to support streaming call for efficiency.
   class HealthCheckContext {
    public:
-    HealthCheckContext(std::shared_ptr<GcsHealthCheckManager> manager,
-                       std::shared_ptr<grpc::Channel> channel,
+    HealthCheckContext(const std::shared_ptr<GcsHealthCheckManager> &manager,
```

Comment:
I think it's better to pass by value here to make it clear that we are extending the lifetime of the shared_ptr.